sdks/python/src/opik/integrations/dspy/callback.py

```diff
@@ -42,12 +43,18 @@ def __init__(
 
         self._opik_client = opik_client.get_client_cached()
 
+    def _skip_tracking(self) -> bool:
+        return not is_tracing_active()
+
     def on_module_start(
         self,
         call_id: str,
         instance: Any,
         inputs: Dict[str, Any],
     ) -> None:
+        if self._skip_tracking():
```

Comment:
Let's only check it right before `self._opik_client.trace()` and `self._opik_client.span()` calls. Due to context safety reasons. Such changes in callback integrations require testing, so I think blocking the data right before calling `Opik` methods is the safest option for now we could go with.